### DISCUSSION TITLE
Custom Netbox Script to log Created or Deleted Assets

### DISCUSSION THREAD

**Discussion by Lurinoxi** [netbox-community/netbox#17022] [Help Wanted!]
Hello Community, i wanted to know if its possible to make a Custom Script which logs all newly created or deleted Inventory Assets, because we have to send them to our Finance team for them to Log them in their system, i am a "Noob" at scripting so i am asking if anyone has done this / is willing to help me! 

Thanks in Advance!

**Comment by jsenecal** [comment]
Define "Newly"? But yes this is very possible and somewhat easy to achieve.

**Reply by Lurinoxi** [reply]
Newly as in if i Create a Iteam now it should Log it then send a mail to the finance team, and if i delete this item (not update) it should do the same, i am using the Inventory Plugin, dont know if this affects this process

**Reply by jsenecal** [reply]
Ok, for this you need webhooks (to an external system) and this is outside the scope of netbox scripts. This also could be achieved by a plugin using Django Signals. Unfortunately you mentioned not being very good at "scripting", so I would recommend you find some help :)

**Reply by Lurinoxi** [reply]
okay, so i have to make like a .py script which a webhook connects to.. got it! thank you for your input

**Reply by jsenecal** [reply]
Well, its either a plugin, or a web app to handle webhooks from netbox. Good luck !

**Comment by chris240189** [comment]
You could create a custom script that is called by an event rule that is bound to that object type and is triggered only on create and delete events.
However, I'd rather create a report regularly (each month) with data from the netbox change log.

**Reply by Lurinoxi** [reply]
I also considered that, and then i will just manually send a mail each month :) but i will try the automation first !

**Reply by chris240189** [reply]
I wrote a somewhat hacky script that I have scheduled to run each month from netbox sending a csv report with changes via email in which I do that.

**Reply by Lurinoxi** [reply]
oh thats cool! would you mind sharing this script? Or parts of the code?
